app/src/main/java/cm/aptoide/pt/abtesting/experiments/IronSourceInterstitialAdExperiment.java

```diff
@@ -0,0 +1,77 @@
+package cm.aptoide.pt.abtesting.experiments;
+
+import cm.aptoide.pt.AptoideApplicationAnalytics;
+import cm.aptoide.pt.abtesting.ABTestManager;
+import cm.aptoide.pt.abtesting.Experiment;
+import cm.aptoide.pt.ads.IronSourceAdRepository;
+import rx.Observable;
+import rx.Scheduler;
+import rx.Single;
+
+public class IronSourceInterstitialAdExperiment {
+
+  private final String EXPERIMENT_ID = "ASV-1187-IronSource-Video-Interstitial";
+
+  private ABTestManager abTestManager;
+  private Scheduler scheduler;
+  private IronSourceAdRepository ironSourceAdRepository;
+  private AptoideApplicationAnalytics aptoideApplicationAnalytics;
+
+  public IronSourceInterstitialAdExperiment(ABTestManager abTestManager, Scheduler scheduler,
+      IronSourceAdRepository ironSourceAdRepository,
+      AptoideApplicationAnalytics aptoideApplicationAnalytics) {
+    this.abTestManager = abTestManager;
+    this.scheduler = scheduler;
+    this.ironSourceAdRepository = ironSourceAdRepository;
+    this.aptoideApplicationAnalytics = aptoideApplicationAnalytics;
+  }
+
+  public Observable<Experiment> loadInterstitial() {
+    return abTestManager.getExperiment(EXPERIMENT_ID)
+        .observeOn(scheduler)
```

Comment:
We should be careful with this because we are forcing the mainthread but we are possibly not only touching the view after this (calls to analytics for instance). Don't think we should quick fix this since it is an ab testing experiment.